compsequential.go

```diff
@@ -184,50 +184,38 @@ func getChannelOrErrorSequential[T any](
 	routers []*SequentialRouter,
 	f func(context.Context, routing.Routing) (<-chan T, error),
 	shouldStop func() bool,
-) (chan T, error) {
+) chan T {
 	chanOut := make(chan T)
-	var chans []<-chan T
-	var cancels []context.CancelFunc
-
-	for _, router := range routers {
-		ctx, cancel := context.WithTimeout(ctx, router.Timeout)
-		rch, err := f(ctx, router.Router)
-		if err != nil &&
-			!errors.Is(err, routing.ErrNotFound) &&
-			!router.IgnoreError {
-			cancel()
-			return nil, err
-		}
-
-		cancels = append(cancels, cancel)
-		chans = append(chans, rch)
-	}
 
 	go func() {
-		for i := 0; i < len(chans); i++ {
-			if chans[i] == nil {
-				cancels[i]()
-				continue
+		for _, router := range routers {
+			ctx, cancel := context.WithTimeout(ctx, router.Timeout)
+			rch, err := f(ctx, router.Router)
+			if err != nil &&
+				!errors.Is(err, routing.ErrNotFound) &&
+				!router.IgnoreError {
+				cancel()
+				break
```

Comment:
Do you think it makes sense to log these attempts and errors at info/debug level so that someone can debug what's happening?  (if so, we can follow up with this in a separate PR)

Reply:
yep, it would make sense